frontend/src/components/ClassBlocks/ClassBlock.tsx

```diff
@@ -16,20 +16,23 @@ const Block = styled(Card)<any>`
   flex-direction: row;
 `;
 
-const Indent = styled.div`
+const Indent = styled.div<any>`
   width: 20px;
-  background-color: rgba(173, 198, 255, 0.9);
+  background-color: ${props =>
+    props.warning ? "rgba(175, 50, 50, 0.9)" : "rgba(173, 198, 255, 0.9)"};
 `;
 
-const ClassBlockBody = styled.div`
-  background-color: rgba(173, 198, 255, 0.3);
+const ClassBlockBody = styled.div<any>`
+  background-color: ${props =>
+    props.warning ? "rgba(216, 86, 86, 0.9)" : "rgba(173, 198, 255, 0.3)"};
   padding-left: 8px;
   flex: 1;
 `;
 
 interface ClassBlockProps {
   class: DNDScheduleCourse;
   index: number;
+  warning: boolean;
```

Comment:
Could you change the type of warning to `string` instead, and store the message here, if it has one?